static/src/javascripts/projects/common/modules/discussion/comments.js

```diff
@@ -0,0 +1,588 @@
+
+                $.create(
+                    `<li class="${this.getClass('showReplies', true)}"></li>`
+                )
+                    .append($btn)
+                    .appendTo($('.d-thread--responses', elem));
+            }
+        });
+    }
+
+    fetchComments(options: Object = {}): Promise<mixed> {
+        const url = `/discussion/${options.comment
+            ? `comment-context/${options.comment}`
+            : this.options.discussionId}.json`;
+        let orderBy = options.order || this.options.order;
+        let promise;
+        const ajaxParams = { mode: 'cors' };
+
+        if (orderBy === 'recommendations') {
+            orderBy = 'mostRecommended';
+        }
+
+        const queryParams: Object = {
+            orderBy,
+            pageSize: options.pagesize || this.options.pagesize,
+            displayThreaded: this.options.threading !== 'unthreaded',
+            commentsClosed: options.commentsClosed,
+        };
+
+        if (options.page) {
+            queryParams.page = options.page;
+        }
+
+        if (!options.comment && this.options.threading === 'collapsed') {
+            queryParams.maxResponses = 3;
+        }
+
+        if (this.isAllPageSizeActive()) {
+            promise = new WholeDiscussion({
+                discussionId: this.options.discussionId,
+                orderBy: queryParams.orderBy,
+                displayThreaded: queryParams.displayThreaded,
+                maxResponses: queryParams.maxResponses,
+                commentsClosed: queryParams.commentsClosed,
+            })
+                .loadAllComments()
+                .catch(() => {
+                    this.wholeDiscussionErrors = true;
+                    queryParams.pageSize = 100;
+
+                    return fetchJson(
+                        `${url}?${constructQuery(queryParams)}`,
+                        ajaxParams
+                    );
+                });
+        } else {
+            // It is possible that the user has chosen to view all comments,
+            // but the WholeDiscussion module has failed. Fall back to 100 comments.
+            if (queryParams.pageSize === 'All') {
+                queryParams.pageSize = 100;
+            }
+
+            promise = fetchJson(
+                `${url}?${constructQuery(queryParams)}`,
+                ajaxParams
+            );
+        }
+
+        return promise.then(resp => this.renderComments(resp));
+    }
+
+    handlePickClick(e: Event): void {
+        e.preventDefault();
+
+        const target: HTMLElement = (e.target: any);
+        const commentId = target.getAttribute('data-comment-id');
+        const $thisButton = $(target);
+        const promise =
+            $thisButton[0].getAttribute('data-comment-highlighted') === 'true'
+                ? this.unPickComment
+                : this.pickComment;
+
+        if (commentId) {
+            promise.call(this, commentId, $thisButton).catch(resp => {
```

Comment:
Is it necessary to pass a new context to `promise`? Why not `promise(commentId, $thisButton)`